### ISSUE TITLE
Autosubmit doesn't work on ea.com

### ISSUE THREAD

**Issue by chrboe** [dannyvankooten/browserpass#201]
### General information

* Operating system + version: Gentoo Linux
* Browser + version: Firefox
* If installed an official release, put a release version: 
* If built from sources, put a commit id (`$ git describe --always`): 01c6253
* Browserpass extension version as reported by your browser: 2.0.9

---

### Exact steps to reproduce the problem
1. Navigate to ea.com
2. Autofill

### What should happen?
Login

### What happened instead?
Info gets filled, login button isn't pressed

### Extra
It's a span inside of a hyperlink, idk what shouldn't be working here 🤔 

PS: sorry for my recent lack of activity regarding this project, I've been lurking and skimming over conversations but haven't had time to contribute to discussions.

**Comment by maximbaz** [comment]
Hey there, I had a quick look, the reason the button is not pressed is because their markup is awful and it's not actually a button, but a span inside a hyperlink as you've correctly pointed out.

We identify submit buttons as either `input[type=submit]` or `button` with certain markers (see [here](https://github.com/dannyvankooten/browserpass/blob/76805fa2a131c4443c1208ca40b360896bc4dc43/chrome/inject.js#L33-L71)), and I'm hesitant to extend the list with the hyperlinks and/or spans, primarily because this has a potential to break other websites which do have a proper markup.

If there is a way to submit a feedback on that website, I would suggest doing so and telling them to read some kind of "HTML 101" book and learn how to make login forms 🙂

I kind of predicted that some websites will have a submit button that is unidentifiable or unclickable, so there is a fallback mechanism: the cursor stays on the password field and hitting Enter submits the form - this seems to work on that website, at least you aren't required to look for your mouse 🙂

In summary, we _can_ add hyperlinks to the list of submit buttons, but I'm tending to suggest to close this as won't fix due to the reasons above, unless you feel strongly about this.

**Comment by chrboe** [comment]
I absolutely believe EA is at fault here (their entire site is littered with unspeakable atrocities, not the least of which is their 20 character **upper** limit for passwords). I'll try to contact them regarding this, though I have to say I hardly even expect a reply.

Feel free to "won't fix" this, I'll follow up here if anything significant happens.

As always, thanks for the excellent response time!
